pkg/scheduler/plugins/elasticquota/core/group_quota_manager.go

```diff
@@ -0,0 +1,487 @@
+		if quotaName == extension.SystemQuotaName || quotaName == extension.DefaultQuotaName {
+			continue
+		}
+		allQuotaTopoNodes[quotaName] = NewQuotaTopoNode(quotaInfo)
+	}
+	//get newCreated nodes
+	for _, quotaInfo := range updateGroups {
+		if _, exist := allQuotaTopoNodes[quotaInfo.Name]; !exist {
+			topoNode := NewQuotaTopoNode(quotaInfo)
+			allQuotaTopoNodes[quotaInfo.Name] = topoNode
+		}
+	}
+
+	// build tree according to the parGroupName
+	for _, topoNode := range allQuotaTopoNodes {
+		if topoNode.name == extension.RootQuotaName {
+			continue
+		}
+		parQuotaTopoNode := allQuotaTopoNodes[topoNode.quotaInfo.ParentName]
+		topoNode.parQuotaTopoNode = parQuotaTopoNode
+		parQuotaTopoNode.AddChildGroupQuotaInfo(topoNode)
+	}
+
+	return allQuotaTopoNodes
+}
+
+// ResetAllGroupQuotaNoLock no need to lock gqm.lock
+func (gqm *GroupQuotaManager) resetAllGroupQuotaNoLock(newAllQuotaTopoNodes map[string]*QuotaTopoNode) {
+	childRequestMap, childUsedMap := make(quotaResMapType), make(quotaResMapType)
+	for quotaName, topoNode := range newAllQuotaTopoNodes {
+		if quotaName == extension.RootQuotaName {
+			continue
+		}
+		topoNode.quotaInfo.lock.Lock()
+		if !topoNode.quotaInfo.IsParent {
+			childRequestMap[quotaName] = topoNode.quotaInfo.CalculateInfo.Request.DeepCopy()
+			childUsedMap[quotaName] = topoNode.quotaInfo.CalculateInfo.Used.DeepCopy()
+		}
+		topoNode.quotaInfo.clearForResetNoLock()
+		topoNode.quotaInfo.lock.Unlock()
+	}
+
+	// clear old structure
+	gqm.runtimeQuotaCalculatorMap = make(map[string]*RuntimeQuotaCalculator)
+	// reset structure
+	gqm.runtimeQuotaCalculatorMap[extension.RootQuotaName] = NewRuntimeQuotaCalculator(extension.RootQuotaName)
+	gqm.runtimeQuotaCalculatorMap[extension.RootQuotaName].SetClusterTotalResource(gqm.totalResourceExceptSystemAndDefaultUsed)
+	rootNode := newAllQuotaTopoNodes[extension.RootQuotaName]
+	gqm.resetAllGroupQuotaRecursiveNoLock(rootNode)
+	gqm.updateResourceKeyNoLock()
+
+	// subGroup's topo relation may change; refresh the request/used from bottom to up
+	for quotaName, topoNode := range newAllQuotaTopoNodes {
+		gqm.quotaTopoNodeMap[quotaName] = topoNode
+		if !topoNode.quotaInfo.IsParent {
+			gqm.updateGroupDeltaRequestNoLock(quotaName, childRequestMap[quotaName])
+			gqm.updateGroupDeltaUsedNoLock(quotaName, childUsedMap[quotaName])
+		}
+	}
+
+	for quotaName, topoNode := range newAllQuotaTopoNodes {
+		if quotaName != extension.RootQuotaName && !topoNode.quotaInfo.IsParent {
+			gqm.refreshRuntimeNoLock(quotaName)
+		}
+	}
+}
+
+// ResetAllGroupQuotaRecursiveNoLock no need to lock gqm.lock
+func (gqm *GroupQuotaManager) resetAllGroupQuotaRecursiveNoLock(rootNode *QuotaTopoNode) {
+	childGroupQuotaInfos := rootNode.GetChildGroupQuotaInfos()
+	for subName, topoNode := range childGroupQuotaInfos {
+		gqm.runtimeQuotaCalculatorMap[subName] = NewRuntimeQuotaCalculator(subName)
+		// equal, not deepCopy
+		gqm.quotaInfoMap[subName] = topoNode.quotaInfo
```

Comment:
as annotation above, "gqm.quotaInfoMap[subName] = topoNode.quotaInfo" can be killed